code/modules/organs/organ_internal.dm

```diff
@@ -255,24 +245,49 @@
 	removed_type = /obj/item/organ/kidneys
 	robotic_type = /obj/item/organ/kidneys/prosthetic
 	organ_id = ORGAN_KIDNEYS
+	var/current_medicine_count = 0 //Tracks the number of reagent/medicine datums we currently have
+	var/current_medicine_cap = 5 //How many drugs we can take before they overwhelm us. Decreases with damage
+
+/datum/internal_organ/kidneys/New(mob/living/carbon/carbon_mob)
+	. = ..()
+	RegisterSignal(carbon_mob.reagents, COMSIG_NEW_REAGENT_ADD, .proc/increase_medicines)
+	RegisterSignal(carbon_mob.reagents, COMSIG_REAGENT_DELETED, .proc/decrease_medicines)
+
+/datum/internal_organ/kidneys/clean_owner()
+	UnregisterSignal(owner.reagents, list(COMSIG_NEW_REAGENT_ADD, COMSIG_REAGENT_DELETED))
+	return ..()
+
+/datum/internal_organ/kidneys/proc/increase_medicines(datum/source, reagent_type, amount)
+	SIGNAL_HANDLER
+	if(ispath(reagent_type, /datum/reagent/medicine))
+		current_medicine_count++
+		if(current_medicine_count == current_medicine_cap + 1)
+			to_chat(owner, span_warning("All the different drugs in you are starting to make you feel off..."))
+
+/datum/internal_organ/kidneys/proc/decrease_medicines(datum/source, reagent_type)
+	SIGNAL_HANDLER
+	if(ispath(reagent_type, /datum/reagent/medicine))
+		current_medicine_count--
+		if(current_medicine_count == current_medicine_cap)
+			to_chat(owner, span_notice("You don't feel as overwhelmed by all the drugs any more."))
+
+/datum/internal_organ/kidneys/set_organ_status()
+	. = ..()
+	if(!.)
+		return
+	current_medicine_cap = initial(current_medicine_cap) - 2 * organ_status
 
 /datum/internal_organ/kidneys/process()
 	..()
 
-	// Coffee is really bad for you with busted kidneys.
-	// This should probably be expanded in some way, but fucked if I know
-	// what else kidneys can process in our reagent list.
-	var/datum/reagent/coffee = locate(/datum/reagent/consumable/drink/coffee) in owner.reagents.reagent_list
-	if(coffee)
-		if(organ_status == ORGAN_BRUISED)
-			owner.adjustToxLoss(0.1 * PROCESS_ACCURACY)
-		else if(organ_status == ORGAN_BROKEN)
-			owner.adjustToxLoss(0.3 * PROCESS_ACCURACY)
-
-	if(organ_status == ORGAN_BRUISED && prob(25))
-		owner.adjustToxLoss(0.1 * (damage/3))
-	else if(organ_status == ORGAN_BROKEN && prob(50))
-		owner.adjustToxLoss(0.2 * (damage/3))
+	if(owner.reagents.has_reagent(/datum/reagent/water))
+		return //Hydration is good for your kidneys. Shame it purges medicines.
+
+	var/overflow = current_medicine_count - current_medicine_cap
+	if(overflow > 0)
+		owner.set_drugginess(3)
+		if(prob(overflow * (organ_status + 1) * 10))
+			owner.Confused(2 SECONDS * (organ_status + 1))
```

Comment:
What is confused doing again ?

Reply:
Chance for random movement when you try to walk. Means there's no real drawbacks if you're standing still, just the druggy overlay.